src/user-safe/UserSafe.sol

```diff
@@ -0,0 +1,678 @@
+     */
+    function transferWeETHToDebtManager(
+        uint256 amount
+    ) external onlyEtherFiCashDebtManager {
+        _checkSpendingLimit(weETH, amount);
+
+        if (
+            amount + blockedFundsForWithdrawal[weETH] >
+            IERC20(weETH).balanceOf(address(this))
+        ) revert InsufficientBalance();
+
+        IERC20(weETH).safeTransfer(msg.sender, amount);
+        emit TransferWeETHAsCollateral(amount);
+    }
+
+    /**
+     * @notice Function to swap WeETH to USDC and transfer it to EtherFiCash Safe.
+     * @dev Can only be called by the EtherFiCash Safe.
+     * @param amountWeETHToSwap Amount of WeETH to swap.
+     * @param minUsdcAmount Min amount of USDC to receive from the swap.
+     * @param amountUsdcToSend Amount of USDC to send to the EtherFiCash Safe.
+     * @param swapData Swap data received from the swapper API.
+     */
+    function swapAndTransfer(
+        uint256 amountWeETHToSwap,
+        uint256 minUsdcAmount,
+        uint256 amountUsdcToSend,
+        bytes calldata swapData
+    ) external onlyEtherFiCashSafe {
+        if (
+            amountWeETHToSwap + blockedFundsForWithdrawal[weETH] >
+            IERC20(weETH).balanceOf(address(this))
+        ) revert InsufficientBalance();
+
+        uint256 returnAmount = _swapWeETHToUsdc(
+            amountWeETHToSwap,
+            minUsdcAmount,
+            swapData
+        );
+        if (amountUsdcToSend > returnAmount)
+            revert AmountGreaterThanUsdcReceived();
+
+        _checkSpendingLimit(usdc, amountUsdcToSend);
+        IERC20(usdc).safeTransfer(msg.sender, amountUsdcToSend);
+
+        emit SwapTransferForSpending(amountWeETHToSwap, amountUsdcToSend);
+    }
+
+    function _getSpendingLimitRenewalTimestamp(
+        uint64 startTimestamp,
+        SpendingLimitTypes spendingLimitType
+    ) internal pure returns (uint64 renewalTimestamp) {
+        if (spendingLimitType == SpendingLimitTypes.Daily)
+            return startTimestamp + 24 * 60 * 60;
+        else if (spendingLimitType == SpendingLimitTypes.Weekly)
+            return startTimestamp + 7 * 24 * 60 * 60;
+        else if (spendingLimitType == SpendingLimitTypes.Monthly)
+            return startTimestamp + 30 * 24 * 60 * 60;
+        else return startTimestamp + 365 * 24 * 60 * 60;
+    }
+
+    function _swapWeETHToUsdc(
+        uint256 amount,
+        uint256 minUsdcAmount,
+        bytes calldata swapData
+    ) internal returns (uint256) {
+        IERC20(weETH).safeTransfer(address(_swapper), amount);
+        return _swapper.swap(weETH, usdc, amount, minUsdcAmount, swapData);
+    }
+
+    function _setSpendingLimit(
+        uint8 spendingLimitType,
+        uint256 limitInUsd
+    ) internal {
+        _spendingLimit = SpendingLimitData({
+            spendingLimitType: SpendingLimitTypes(spendingLimitType),
+            renewalTimestamp: _getSpendingLimitRenewalTimestamp(
+                uint64(block.timestamp),
```

Comment:
we may want to specify the specific renewal time instead of current time.

e.g., i wanna set it to be Sunday midnight in my timezone

Reply:
Okay so we want a start time as well? I think we can use the incoming spending limit for that.

Reply:
i voted for removing the `incoming...` thing. so, if that is required to have this feature, I don't think it is necessary

Reply:
Makes sense

Reply:
Discussed with Koray about these things. For eg, we will have 30 day cycles for monthly limits then also we will deviate from the actual monthly cycle because some months are 31 days etc. So we figured just start the time from current timestamp.

Reply:
sg